blueprints/multi-interface-servers.md

```diff
@@ -0,0 +1,670 @@
+	 * address.
+	 */
+	ipAddresses: Array<IPAddress> & {0: IPAddress};
+	/**
+	 * Whether or not Traffic Monitor should monitor this particular interface.
+	 */
+	monitor: boolean;
+	/**
+	 * The interface's Maximum Transmission Unit.
+	 * If this is 'null' it is assumed that the interface's MTU is not known/is
+	 * irrelevant.
+	 */
+	mtu: 1500 | 9000 | null;
+}
+```
+We don't have a real data model for ATC, so what follows is an approximately
+defined representation of a "server" that is valid within certain contexts. The
+modifications made to it for the purposes of the changes herein proposed should
+be considered valid in those same contexts - whatever they may be.
+
+```typescript
+interface Server {
+
+	cachegroup?:  string | null;
+	cachegroupId: number;
+	cdnId:        number;
+	cdnName?:     string | null;
+	domainName:   string;
+	guid?:        string | null;
+	hostName:     string;
+	httpsPort?:   number | null;
+	id?:          number;
+
+	// ILO things aren't being moved in because they also require a
+	// username/password that aren't used by any other interface type.
+	iloIpAddress?: string | null;
+	iloIpGateway?: string | null;
+	iloIpNetmask?: string | null;
+	iloPassword?:  string | null;
+	iloUsername?:  string | null;
+
+	/**
+	 * New field containing additional interfaces.
+	 * It is illegal for a server to not have at least one interface.
+	 */
+	interfaces: Array<Interface> & {0: Interface};
+	lastUpdated?:  string;
+
+	// notice not here: mgmtIpAddress, mgmtIpGateway, mgmtIpNetmask
+	// these will be moved into the interfaces property
+
+	offlineReason:   string;
+	physLocation?:   string | null;
+	physLocationId:  number;
+	profile?:        string | null;
+	profileDesc?:    string | null;
+	profileId:       number;
+	revalPending?:   boolean;
+	rack?:           string | null;
+	routerHostName?: string | null;
+	routerPortName?: string | null;
+	status?:         string | null;
+	statusId:        number;
+	tcpPort:         number;
+	type?:           string | null;
+	typeId:          number;
+	updPending?:     boolean;
+	xmppId?:         string | null;
+	xmppPassword?:   string | null;
+}
+```
+
+`/monitoring` payloads and CDN Snapshots are purposely un-versioned and
+un-modeled, so this section will make no attempts to define the model changes to
+those objects.
+
+## Impacted Components
+The only components that should need to be modified to accommodate this feature
+are Traffic Ops, Traffic Portal, and Traffic Monitor.
+
+### Traffic Ops Impact
+
+#### Database
+A new table will need to be introduced to contain interface information:
+`interface`, which is described below.
+```text
+              Table "public.interface"
+ Column  |  Type   | Collation | Nullable | Default
+---------+---------+-----------+----------+---------
+ name    | text    |           | not null |
+ server  | bigint  |           | not null |
+ monitor | boolean |           | not null |
+ mtu     | integer |           |          | 1500
```

Comment:
if we're going to only support two values and null, why use an integer type over an enum or domain (with appropriate check constraint)?

Reply:
That's probably better, but to be honest I'm still not sure what MTU actually is or if 1500 and 9000 are the only values that ever can or will be possible for network interface devices. I was sort of hoping someone would call me out on that assertion if it was wrong.

Reply:
MTU is Maximum Transmission Unit and has to do with the size of Layer 2 Ethernet frames.  1500 is more or less the default, but if you can get 9000 beginning to end you should. https://en.wikipedia.org/wiki/Jumbo_frame

Super jumbo frames technically exist, but I've not run across them except in router configs.

Reply:
There doesn't appear to be a spec for "super jumbo frames", but maybe I should add the "baby giant frames" value (1600).

Also, what're they gonna do when "super jumbo" *is* standardized and they want something even bigger? "Giga Frames"? "Maximum Over-Frames"?

Reply:
There are many different values that you _could__ set for MTU.  1500 and 9000 are the most common, there's a 4500ish one that is pretty common too.  I  DO NOT think this should be an enum.

Reply:
Well then should we be restricting it at all? Maybe just >= 1500? Because currently TP will only let you set 1500 or 9000, which is why I chose those restrictions in the first place, but if it could theoretically be set to anything then maybe TP should just set a reasonable default but not restrict allowable values there, or in the API or database?

Reply:
Per https://www.ietf.org/rfc/rfc2460.txt we should probably constrain it to _at least_ 1280 or higher, but TP should probably still show 1500 and 9000 as the available options until we know we need different sizes.

Reply:
I went with >1280